### PR TITLE
fix bug in semi sync

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
after slave send ack to master, master will not reply
this is fix for https://github.com/siddontang/go-mysql/issues/176

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
I think we also need a overtime control when read package @siddontang

File: replication/binlogsyncer.go [hunk 1]
```diff
@@ -514,10 +514,7 @@ func (b *BinlogSyncer) replySemiSyncACK(p Position) error {
 		return errors.Trace(err)
 	}
 
-	_, err = b.c.ReadOKPacket()
-	if err != nil {
-	}
-	return errors.Trace(err)
+	return nil
```
Comment (siddontang) [inline replication/binlogsyncer.go]: I remember that I referred old manual and wrote the code here.

where can we know the master will not reply?

**Comment by WangXiangUSTC** [comment]
I find this will  hang on in readOKPackage function, because slave can't read any data after send ack to master.

And I can't find any code about slave read ack from master in mysql source code:
https://github.com/mysql/mysql-server/blob/71f48ab393bce80a59e5a2e498cd1f46f6b43f9a/plugin/semisync/semisync_slave.cc#L99

**Comment by siddontang** [comment]
Cool, LGTM

**Pull request merged**